extensions/amp-app-banner/0.1/amp-app-banner.js

```diff
@@ -381,7 +381,7 @@ export class AmpAndroidAppBanner extends AbstractAppBanner {
 
   /** @private */
   getAndroidIntentForUrl_(appId) {
-    const canonicalUrl = documentInfoFor(this.win).canonicalUrl;
+    const canonicalUrl = documentInfoForDoc(this.element).canonicalUrl;
```

Comment:
Yeah. We went back and force on it. Basically, it starts with the context element which points itself directly at a doc. E.g. `documentInfoForDoc(ampdocService.findAmpDocForElement(this.element))`. Constantly doing this was just two much. So it's hidden behind that API.